### PR TITLE
Change to GitHub actions

### PR DESCRIPTION
**Pull request by wxtim**
## Issue

Tunebook hasn't been building on Travis

## Work undertaken

- [x] Moved to using Github actions.

## How to test

Trial Deployed to https://wxtim.github.io/bristolfolksoc.github.io/tunes.html

## Other

@MorrisNerd2  I'm not sure whether you might want to look at https://github.com/wxtim/bristolfolksoc.github.io/runs/6747568422?check_suite_focus=true  and look at > create tunebook. I can't say for sure whether it's complaining about files because the files are wrong or because the program is - although when I use abcm2ps it's usually me. Looking quickly at "Vandals of Hammerwich"  (just picked one at random) and [the abc standard](https://abcnotation.com/wiki/abc:standard:v2.1#chords_and_unisons) I think that you might have failed to close some brackets around notes in chords:

```diff
 P: Song A
-"G/A" [F4DA, [E4CA,|"D"A,2 D2 F2 D2|...
+"G/A" [F4DA,] [E4CA,]|"D"A,2 D2 F2 D2|...
 P: Normal A

```

### CODE REVIEW

File: travis/deploy.sh
Comment (wxtim) [inline travis/deploy.sh]: Needs change before final merge.

```suggestion
git remote add origin-pages https://git@github.com/bristolfolksoc/bristolfolksoc.github.io.git > /dev/null 2>&1
```

File: travis/deploy.sh
Comment (wxtim) [inline travis/deploy.sh]: This forces the bash script to fail if something goes wrong!

File: .github/workflows/buildtunebook.yaml
Comment (wxtim) [inline .github/workflows/buildtunebook.yaml]: Before merge

```suggestion
          git remote add origin-pages https://git@github.com/bristolfolksoc/bristolfolksoc.github.io.git > /dev/null 2>&1
```

File: tunes/Delvin House.abc
Comment (wxtim) [inline tunes/Delvin House.abc]: a line at the end of a file is good practice @MorrisNerd2

File: .github/workflows/buildtunebook.yaml
Comment (wxtim) [inline .github/workflows/buildtunebook.yaml:33]: @Nick-Pearson In theory using sudo creates a vulnerability to change, however I think these very old scripts are being maintained well - they seem the basis of far too many other ABC programs. I've left the sub-modules for safety's sake, lest we need to go back to the old method (although in theory the VC system should make this academic).

Does this seem reasonable to you?

File: tunes/Delvin House.abc
Comment (Nick-Pearson) [reply tunes/Delvin House.abc]: iirc one of these libraries we use for the book creation is also able to lint abc files which gives the option of checking the files in gh actions as well

File: .github/workflows/buildtunebook.yaml
Comment (Nick-Pearson) [inline .github/workflows/buildtunebook.yaml:50]: Will this be run on a pull request?

File: .github/workflows/buildtunebook.yaml
Comment (Nick-Pearson) [reply .github/workflows/buildtunebook.yaml:33]: Yes seems fair enough to me

File: tunes/DingDongMerrilyonHigh.abc
Comment (Nick-Pearson) [inline tunes/DingDongMerrilyonHigh.abc]: this is read by generate.js as the number of bars

**Comment by Nick-Pearson** [comment]
The latex tunebook pdfs rely on this very Travis specific dockerfile https://github.com/Strauman/travis-latexbuild
We will need to get a github actions equivalent (perhaps that should be done separately to this pr?)

**Comment by MorrisNerd2Again** [comment]
Last one: Yeah let's do it as a separate one.

File: .github/workflows/buildtunebook.yaml
Comment (wxtim) [reply .github/workflows/buildtunebook.yaml:50]: Yes - it only wants to be on merge, doesn't it!

File: tunes/Delvin House.abc
Comment (wxtim) [reply tunes/Delvin House.abc]: I created a separate issue for an on PR change at https://github.com/bristolfolksoc/bristolfolksoc.github.io/issues/22

File: tunes/DingDongMerrilyonHigh.abc
Comment (wxtim) [reply tunes/DingDongMerrilyonHigh.abc]: This is a rebase error.

**Comment by wxtim** [comment]
I've got some changes to make (this should only run to merge with dev branch I think). I'll do that out of work time.

File: .github/workflows/buildtunebook.yaml
Comment (wxtim) [reply .github/workflows/buildtunebook.yaml:50]: I've changed it to 
```
on:
  push:
    branches:
      - development
```

Which is functionally equivalent to on merge (which doesn't exist) unless the developer has direct push permissions.

**Review by Nick-Pearson — APPROVED** [review]

**Comment by Nick-Pearson** [comment]
Looks good to me
Thanks @wxtim!

**Pull request merged**
